subsys/bluetooth/audio/cap_commander.c

```diff
@@ -77,18 +82,314 @@ int bt_cap_commander_discover(struct bt_conn *conn)
+		}
+
+		if (client == NULL) {
+			LOG_DBG("Invalid param->param[%zu].member", i);
+			return false;
+		}
+
+		CHECKIF(start_param->addr.type > BT_ADDR_LE_RANDOM) {
+			LOG_DBG("Invalid address type %u", start_param->addr.type);
+			return false;
+		}
+
+		CHECKIF(start_param->adv_sid > BT_GAP_SID_MAX) {
+			LOG_DBG("param->param[%zu]->adv_sid is larger than %d", i, BT_GAP_SID_MAX);
+			return false;
+		}
+
+		CHECKIF(!IN_RANGE(start_param->pa_interval, BT_GAP_PER_ADV_MIN_INTERVAL,
+				  BT_GAP_PER_ADV_MAX_INTERVAL)) {
+			LOG_DBG("param->param[%zu]->pa_interval is out of range", i);
+			return false;
+		}
+
+		CHECKIF(start_param->broadcast_id > BT_AUDIO_BROADCAST_ID_MAX) {
+			LOG_DBG("param->param[%zu]->broadcast_id is larger than %u", i,
+				BT_AUDIO_BROADCAST_ID_MAX);
+			return false;
+		}
+
+		CHECKIF(start_param->num_subgroups == 0) {
+			LOG_DBG("param->param[%zu]->num_subgroups is 0", i);
+			return false;
+		}
+
+		CHECKIF(start_param->num_subgroups > CONFIG_BT_BAP_BASS_MAX_SUBGROUPS) {
+			LOG_DBG("Too many subgroups %u/%u", start_param->num_subgroups,
+				CONFIG_BT_BAP_BASS_MAX_SUBGROUPS);
+
+			return false;
+		}
+
+		CHECKIF(start_param->subgroups == NULL) {
+			LOG_DBG("param->param[%zu]->subgroup is NULL", i);
+			return false;
+		}
+
+		for (size_t j = 0U; j < start_param->num_subgroups; j++) {
+			struct bt_bap_bass_subgroup *param_subgroup = &start_param->subgroups[j];
+
+			CHECKIF(!valid_bis_syncs(param_subgroup->bis_sync)) {
+				LOG_DBG("param->param[%zu].subgroup[%zu].bis_sync is invalid %u", i,
+					j, param_subgroups->bis_sync);
+			}
+
+			CHECKIF((total_bis_sync & param_subgroups->bis_sync) != 0) {
+				LOG_DBG("param->param[%zu].subgroup[%zu].bis_sync is a duplicate "
+					"%u",
+					i, j, param_subgroups->bis_sync);
+			}
+
+			total_bis_sync |= param_subgroups->bis_sync;
+
+			CHECKIF(param_subgroups->metadata_len >
+				CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE) {
+				LOG_DBG("param->param[%zu].subgroup[%zu].metadata_len too long "
+					"%u/%u",
+					i, j, param_subgroups->metadata_len,
+					CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE);
+
+				return false;
+			}
+#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE)
+			CHECKIF(param_subgroups->metadata_len > 0 &&
+				!bt_audio_valid_ltv(param_subgroups->metadata,
+						    param_subgroups->metadata_len)) {
+				LOG_DBG("param->param[%zu].subgroup[%zu].metadata not valid LTV", i,
+					j);
+			}
+#endif
+		}
+
+		for (size_t j = 0U; j < i; j++) {
+			const union bt_cap_set_member *other = &param->param[j].member;
+
+			if (other == member) {
+				LOG_DBG("param->members[%zu] (%p) is duplicated by "
+					"param->members[%zu] (%p)",
+					j, other, i, member);
+				return false;
+			}
+		}
+	}
+
+	return true;
+}
+#endif /* CONFIG_BT_BAP_BROADCAST_ASSISTANT */
+
 int bt_cap_commander_broadcast_reception_start(
 	const struct bt_cap_commander_broadcast_reception_start_param *param)
 {
+#if defined(CONFIG_BT_BAP_BROADCAST_ASSISTANT)
```

Comment:
```suggestion

int bt_cap_commander_broadcast_reception_start(
	const struct bt_cap_commander_broadcast_reception_start_param *param)
{
```

Reply:
The idea was to keep the bt_cap_commander_broadcast_reception_start function available, but I guess noone should call that function anyways unless they have selected BT_BAP_BROADCAST_ASSISTANT

Reply:
Indeed. It's better to get a compile-time error than an runtime error